src/Symfony/Component/Form/Extension/Core/Type/FileType.php

```diff
@@ -12,15 +12,40 @@
 namespace Symfony\Component\Form\Extension\Core\Type;
 
 use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
 use Symfony\Component\Form\FormInterface;
 use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\Options;
 use Symfony\Component\OptionsResolver\OptionsResolver;
 
 class FileType extends AbstractType
 {
     /**
      * {@inheritdoc}
      */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        if ($options['multiple']) {
+            $builder->addEventListener(FormEvents::PRE_SUBMIT, function (FormEvent $event) {
+                $form = $event->getForm();
+                $data = $event->getData();
+
+                // submitted data for an input file (not required) without choosing any file
+                if (array(null) === $data) {
```

Comment:
@ro0NL : There is no way to submit `array(null, <file>, null, null)` for a single `FileType` with `multiple` set to `true`. You have just three options: 

Select `<input type="file" multiple="true">`:
1- one file (`array(<UploadedFile>)`)
2- many files (`array(<UploadedFile>, <UploadedFile>, ...)`)
3- none file (`array(null)`)

This is about the 3rd option and normalize the empty data to `array()` when this happen.

> So if many inputs are submitted, and only few files are uploaded, we get `array(null, <file>, null, null)` for instance. And as the data model is a collection, i guess `array(<file>)` is actually expected.

I guess you are talking about a `CollectionType` with `entry_type` set to `FileType::class` (not multiple) ? anyway: `array(null, <file>, null, null)` each of them is assigned to a  `FileType`.